### ISSUE TITLE
super => Class.call.apply(null) => "object is not a function" in 2.4.10

### ISSUE THREAD

**Issue by timoxley** [6to5/6to5#382]
6to5@2.4.10 is generating `ClassName.call.apply(null, ...)` in classes using `super` which creates the error: 

`Uncaught TypeError: object is not a function`

This didn't occur in 2.4.8, and 2.4.9 doesn't install (`npm ERR! version not found: acorn-6to5@0.11.1-7`)

This isn't the first time `super` has been broken in 6to5, maybe need more tests around it or something.

## Input

Note this happens with `class B{}` too.

```js
function B() {}

class A extends B {
  constructor(...args) {
    super(...args)
  }
}
```

## 2.4.8
```js
"use strict";

var _slice = Array.prototype.slice;
var _toArray = function (arr) {
  return Array.isArray(arr) ? arr : Array.from(arr);
};

var _inherits = function (child, parent) {
  child.prototype = Object.create(parent && parent.prototype, {
    constructor: {
      value: child,
      enumerable: false,
      writable: true,
      configurable: true
    }
  });
  if (parent) child.__proto__ = parent;
};

function B() {}

var A = (function () {
  var _B = B;
  var A = function A() {
    var args = _slice.call(arguments);

    _B.call.apply(_B, [this].concat(_toArray(args)));
  };

  _inherits(A, _B);

  return A;
})();
```

## 2.4.10
```js
"use strict";

var _slice = Array.prototype.slice;
var _toArray = function (arr) {
  return Array.isArray(arr) ? arr : Array.from(arr);
};

var _inherits = function (child, parent) {
  child.prototype = Object.create(parent && parent.prototype, {
    constructor: {
      value: child,
      enumerable: false,
      writable: true,
      configurable: true
    }
  });
  if (parent) child.__proto__ = parent;
};

function B() {}

var A = (function () {
  var _B = B;
  var A = function A() {
    var args = _slice.call(arguments);

    _B.call.apply(null, [this].concat(_toArray(args)));
  };

  _inherits(A, _B);

  return A;
})();
```

**Comment by sebmck** [comment]
2.4.9 was accidently pushed out. `super` is going to be rewritten because of lack of support for getters etc. Hoping to get the entire Traceur test suite running (only about 50 tests to go, mostly inconsistencies in the testing abstraction layer so they aren't 6to5 transformation issues).

I can't seem to replicate this specific issue either.

**test.js**

```javascript
function B() {
  console.log(arguments);
}

class A extends B {
  constructor(...args) {
    super(...args)
  }
}

var a = new A("a", "b");
```

```sh
$ 6to5-node test.js
{ "0": "a", "1": "b" }
$
```

**Comment by timoxley** [comment]
weird
```
timoxley@yellow ~/Projects
  % cat t.js
function B() {
  console.log(arguments);
}

class A extends B {
  constructor(...args) {
    super(...args)
  }
}

var a = new A("a", "b");
```
```
  % 6to5-node t.js

/Users/timoxley/Projects/t.js:7
    super(...args)
    ^
TypeError: object is not a function
    at new A (/Users/timoxley/Projects/t.js:7:5)
    at Object.<anonymous> (/Users/timoxley/Projects/t.js:11:9)
    at Module._compile (module.js:456:26)
    at Object.loader (/usr/local/lib/node_modules/6to5/lib/6to5/register.js:110:5)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at Object.<anonymous> (/usr/local/lib/node_modules/6to5/bin/_6to5-node:74:12)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)

timoxley@yellow ~/Projects
  % 6to5 --version
2.4.10
timoxley@yellow ~/Projects
  % node -v
v0.10.35
```

**Comment by sebmck** [comment]
It's probably a change I made to my development copy then. I'm currently doing a lot of work dealing with edgecases so I might have changed something around.

**Comment by timoxley** [comment]
Hang on, I'll try 6to5 master,

**Comment by sebmck** [comment]
@timoxley Just checked, it does.

**Comment by sebmck** [comment]
It's fixed on the esnext branch.

**Comment by timoxley** [comment]
ok… hm… Think I should wait for an imminent new release or back down to 2.4.8?

**Comment by sebmck** [comment]
@timoxley Yeah, I'll merge in the esnext branch and disable the two failing tests. I was hoping to get that branch fully merged in today but I got sidetracked in ~500 Traceur tests.

**Comment by timoxley** [comment]
:thumbsup:

**Comment by sebmck** [comment]
Fixed and released as of 2.5.0. I'm going to delay any further releases until Traceur/esnext test suite development is complete.

**Comment by timoxley** [comment]
I can confirm app is working again. All good.
